src/member.rs

```diff
@@ -0,0 +1,57 @@
+//! [`Member`]s and [`MemberId`]s are the building blocks of
+//! [`crate::problem::ProxyProblem`]s and [`crate::solution::ProxySolution`]s.
+use indexmap::map::IndexMap;
+
+use matchmaker::{Category, Student};
+
+use rocket::serde::Deserialize;
```

Comment:
This is a bit of a confusing trap that rocket contains...rocket wanted to re-export all the common dependencies for people to use, but the way macros work is a bit weird...in short, serde's derive macros only work if you have `serde` as a root dependency in your project, or you specify where serde is with the `#[serde(crate = "...")]` attribute (see https://serde.rs/container-attrs.html#crate). 

So despite having `use rocket::serde::Deserialize;`, when you use `#[derive(Deserialize)]`, it's half using the macro from your explicit serde dependency and the other half from rocket's serde dependency. Of course both point to the same thing (otherwise it would blow up pretty badly) but I find it all unnecessarily confusing and I just do `use serde::Deserialize` always.